inc/ajax.class.php

```diff
@@ -116,6 +116,101 @@ static function createModalWindow($name, $url, $options=array() ) {
       }
    }
 
+   /**
+    * Create a side slide panel
+    *
+    * @param string $name    name of the js object
+    * @param array  $options Possible options:
+    *          - title       Title to display
+    *          - position    position (either left or right - defaults to right)
+    *          - display     display or get string? (default true)
+    *
+    * @return void|string (see $options['display'])
+    */
+   static function createSlidePanel($name, $options = []) {
+      global $CFG_GLPI;
+
+      $param = [
+         'title'     => '',
+         'position'  => 'right',
+         'url'       => '',
+         'display'   => true,
+         'icon'      => false,
+         'icon_url'  => false,
+         'icon_txt'  => false
+      ];
+
+      if (count($options)) {
+         foreach ($options as $key => $val) {
+            if (isset($param[$key])) {
+               $param[$key] = $val;
+            }
+         }
+      }
+
+      $out  =  "<script type='text/javascript'>\n";
+      $out .= "$(function() {";
+      $out .= "$('<div id=\'$name\' class=\'slidepanel on{$param['position']}\'><div class=\"header\">" .
+         "<button type=\"button\" class=\"close ui-button ui-widget ui-state-default ui-corner-all ui-button-icon-only ui-dialog-titlebar-close\" title=\"". __('Close') . "\"><span class=\"ui-button-icon-primary ui-icon ui-icon-closethick\"></span><span class=\"ui-button-text\">". __('Close') ."</span></button>";
+
+      if ($param['icon']) {
+         $icon = "<img class=\"icon\" src=\"{$CFG_GLPI['root_doc']}{$param['icon']}\" alt=\"{$param['icon_txt']}\" title=\"{$param['icon_txt']}\"/>";
```

Comment:
Since this will be used into javascript, we have either to escape `"` for the PHP or `'` for the JS